### ISSUE TITLE
Inconsistencies in ADE20K model inference output for the same input image.

### ISSUE THREAD

**Issue by MartaFdez97** [Visual-Attention-Network/SegNeXt#52]
Hello,
I am doing inference with the ade20k model and I am seeing that for the same input, I get a different output. I have saved the data and model variables to be always the same and yet I still have variability. Why does this happen? Also, I have removed the rescale.
My code: 

```
   import pickle

    with open('data.pkl', 'rb') as f:
        data = pickle.load(f)
    with open('model.pkl', 'rb') as f:
        model = pickle.load(f)
    
    # forward the model
    with torch.no_grad():
        result = model(return_loss=False, rescale=False, **data)
    return result
```

**Comment by uyzhang** [comment]
There is a random reasoning part in [ham_head](https://arxiv.org/abs/2109.04553), which may fluctuate slightly, but it exists.

**Comment by MartaFdez97** [comment]
It fluctuates even if the weights are the fixed? I do not get how can It be possible

**Comment by uyzhang** [comment]
Yes, you can know the detailed Implementation with the [code](https://github.com/Visual-Attention-Network/SegNeXt/blob/main/mmseg/models/decode_heads/ham_head.py#L212)

**Comment by RadekZenkl** [comment]
I encountered the same issue where consequent forward passes produce different outputs. I was able to find a workaround where the torch and torch.cuda seeds are set before each individual forward pass. 

```
torch.manual_seed(42)
torch.cuda.manual_seed(42)
b = model(a)
torch.manual_seed(42)
torch.cuda.manual_seed(42)
c = model(a)

print(torch.allclose(b, c, atol=1e-6))  # True if identical within tolerance
```
